compiler/rustc_data_structures/src/marker.rs

```diff
@@ -0,0 +1,268 @@
+cfg_if!(
+    if #[cfg(not(parallel_compiler))] {
+        pub auto trait DynSend {}
+        pub auto trait DynSync {}
+
+        impl<T> DynSend for T {}
+        impl<T> DynSync for T {}
+    } else {
+        #[rustc_on_unimplemented(
+            message = "`{Self}` doesn't implement `DynSend`. \
+            Add it to `rustc_data_structures::marker` or use `IntoDyn` if it's already `Send`",
+            label = "`{Self}` doesn't implement `DynSend`. \
+            Add it to `rustc_data_structures::marker` or use `IntoDyn` if it's already `Send`"
+        )]
+        // Ensure data structures is `Send` if `sync::active()` is true.
+        // `sync::active()` should be checked before using these data structures.
+        // Note: Ensure that the data structure **will not break**
+        // thread safety after being created.
+        //
+        // `sync::active()` should be checked when downcasting these data structures
+        // to `Send` via `FromDyn`.
+        pub unsafe auto trait DynSend {}
+
+        #[rustc_on_unimplemented(
+            message = "`{Self}` doesn't implement `DynSync`. \
+            Add it to `rustc_data_structures::marker` or use `IntoDyn` if it's already `Sync`",
+            label = "`{Self}` doesn't implement `DynSync`. \
+            Add it to `rustc_data_structures::marker` or use `IntoDyn` if it's already `Sync`"
```

Comment:
Why are the message and label duplicated on these traits?

Reply:
I simply imitated the implementation of `Send` and `Sync`. Looks like removing `label` works here.